src/libzippp.cpp

```diff
@@ -530,8 +548,17 @@ bool ZipArchive::addFile(const string& entryName, const string& file) const {
     zip_source* source = zip_source_file(zipHandle, filepath, 0, end);
     if (source!=nullptr) {
         libzippp_int64 result = zip_file_add(zipHandle, entryName.c_str(), source, ZIP_FL_OVERWRITE);
-        if (result>=0) { return true; } 
-        else { zip_source_free(source); } //unable to add the file
+        if (result>=0) { 
+            if(isEncrypted()){
+                if(zip_file_set_encryption(zipHandle,result,encryptionMethod,nullptr)!=0){//unable to encrypt
+                    zip_source_free(source);
+                }else{
+                    return true;
+                }
+            }else{
+                return true;
+                }
+        } else { zip_source_free(source); } //unable to add the file
```

Comment:
Please fix indentation and format:
- space between braces
- line return after opening braces to keep consistency